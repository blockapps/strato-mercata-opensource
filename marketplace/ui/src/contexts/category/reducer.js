import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createCategory:
      return {
        ...state,
        isCreateCategorySubmitting: true,
      };
    case actionDescriptors.createCategorySuccessful:
      return {
        ...state,
        category: action.payload,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.createCategoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.fetchCategory:
      return {
        ...state,
        iscategorysLoading: true,
      };
    case actionDescriptors.fetchCategorySuccessful:
      return {
        ...state,
        categorys: action.payload,
        iscategorysLoading: false,
      };
    case actionDescriptors.fetchCategoryFailed:
      return {
        ...state,
        error: action.error,
        iscategorysLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
