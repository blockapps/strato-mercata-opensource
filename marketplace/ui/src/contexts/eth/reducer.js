import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchETHSTAddress:
      return {
        ...state,
      };
    case actionDescriptors.fetchETHSTAddressSuccessful:
      return {
        ...state,
        ethstAddress: action.payload,
      };
    case actionDescriptors.fetchETHSTAddressFailed:
      return {
        ...state,
      };
    case actionDescriptors.addHash:
      return {
        ...state,
        isAddingHash: true,
      };
    case actionDescriptors.addHashSuccessful:
      return {
        ...state,
        isAddingHash: false,
      };
    case actionDescriptors.addHashFailed:
      return {
        ...state,
        isAddingHash: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
