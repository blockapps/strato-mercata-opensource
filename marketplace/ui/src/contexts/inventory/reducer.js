import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createInventory:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createInventoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchInventory:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUser:
      return {
        ...state,
        isUserInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventoryForUserSuccessful:
      return {
        ...state,
        userInventories: action.payload.data,
        userInventoriesTotal: action.payload.count,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUserFailed:
      return {
        ...state,
        error: action.error,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearch:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySearchSuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearchFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.updateInventory:
      return {
        ...state,
        isinventoryUpdating: true,
      };
    case actionDescriptors.updateInventorySuccessful:
      return {
        ...state,
        inventoryUpdateObject: action.payload,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateInventoryFailed:
      return {
        ...state,
        error: action.error,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateSale:
      return {
        ...state,
        issaleUpdating: true,
      };
    case actionDescriptors.updateSaleSuccessful:
      return {
        ...state,
        saleUpdateObject: action.payload,
        issaleUpdating: false,
      };
    case actionDescriptors.updateSaleFailed:
      return {
        ...state,
        error: action.error,
        issaleUpdating: false,
      };
    case actionDescriptors.listInventory:
      return {
        ...state,
        isListing: true,
      };
    case actionDescriptors.listInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isListing: false,
      };
    case actionDescriptors.listInventoryFailed:
      return {
        ...state,
        error: action.error,
        isListing: false,
      };
    case actionDescriptors.unlistInventory:
      return {
        ...state,
        isUnlisting: true,
      };
    case actionDescriptors.unlistInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isUnlisting: false,
      };
    case actionDescriptors.unlistInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnlisting: false,
      };

     // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.stakeInventory:
      return {
        ...state,
        isStaking: true
      };
    case actionDescriptors.stakeInventorySuccessful:
      return {
        ...state,
        isStaking: false
      };
    case actionDescriptors.stakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isStaking: false
      };
    
    case actionDescriptors.unstakeInventory:
      return {
        ...state,
        isUnstaking: true
      };
    case actionDescriptors.unstakeInventorySuccessful:
      return {
        ...state,
        isUnstaking: false
      };
    case actionDescriptors.unstakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnstaking: false
      };
    
    case actionDescriptors.borrow:
      return {
        ...state,
        isBorrowing: true
      };
    case actionDescriptors.borrowSuccessful:
      return {
        ...state,
        isBorrowing: false
      };
    case actionDescriptors.borrowFailed:
      return {
        ...state,
        error: action.error,
        isBorrowing: false
      };
    
    case actionDescriptors.repay:
      return {
        ...state,
        isRepaying: true
      };
    case actionDescriptors.repaySuccessful:
      return {
        ...state,
        isRepaying: false
      };
    case actionDescriptors.repayFailed:
        return {
          ...state,
          error: action.error,
          isRepaying: false
        };

    case actionDescriptors.getReserve:
        return {
          ...state,
          isReserveLoading: true
        };
    case actionDescriptors.getReserveSuccessful:
        return {
          ...state,
          reserve: action.payload,
          isReserveLoading: false
        };
    case actionDescriptors.getReserveFailed:
        return {
          ...state,
          error: action.error,
          isReserveLoading: false
        };
    
    case actionDescriptors.getAllReserve:
        return {
          ...state,
          isReservesLoading: true
        };
    case actionDescriptors.getAllReserveSuccessful:
        return {
          ...state,
          reserves: action.payload,
          isReservesLoading: false
        };
    case actionDescriptors.getAllReserveFailed:
        return {
          ...state,
          error: action.error,
          isReservesLoading: false
        };
    
    case actionDescriptors.getEscrowForAsset:
        return {
          ...state,
          isEscrowLoading: true
        };
    case actionDescriptors.getEscrowForAssetSuccessful:
        return {
          ...state,
          escrow: action.payload,
          isEscrowLoading: false
        };
    case actionDescriptors.getEscrowForAssetFailed:
        return {
          ...state,
          error: action.error,
          isEscrowLoading: false
        };

    case actionDescriptors.getUserCataRewards:
        return {
          ...state,
          isUserCataRewardsLoading: true
        };
    case actionDescriptors.getUserCataRewardsSuccessful:
        return {
          ...state,
          totalCataReward: action.payload.totalCataReward,
          dailyCataReward: action.payload.dailyCataReward,
          isUserCataRewardsLoading: false
        };
    case actionDescriptors.getUserCataRewardsFailed:
        return {
          ...state,
          error: action.error,
          isUserCataRewardsLoading: false
        };
    
    case actionDescriptors.getOracle:
          return {
            ...state,
            isFetchingOracle: true
          };
    case actionDescriptors.getOracleSuccessful:
          return {
            ...state,
            oracle: action.payload,
            isFetchingOracle: false
          };
    case actionDescriptors.getOracleFailed:
          return {
            ...state,
            error: action.error,
            isFetchingOracle: false
          };    
    // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.resellInventory:
      return {
        ...state,
        isReselling: true,
      };
    case actionDescriptors.resellInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isReselling: false,
      };
    case actionDescriptors.resellInventoryFailed:
      return {
        ...state,
        error: action.error,
        isReselling: false,
      };
    case actionDescriptors.fetchSupportedTokens:
      return {
        ...state,
        isFetchingTokens: true,
      };
    case actionDescriptors.fetchSupportedTokensSuccessful:
      return {
        ...state,
        supportedTokens: action.payload,
        isFetchingTokens: false,
      };
    case actionDescriptors.fetchSupportedTokensFailed:
      return {
        ...state,
        error: action.error,
        isFetchingTokens: false,
      };
    case actionDescriptors.bridgeInventory:
      return {
        ...state,
        isBridging: true,
      };
    case actionDescriptors.bridgeInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isBridging: false,
      };
    case actionDescriptors.bridgeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isBridging: false,
      };
    case actionDescriptors.transferInventory:
      return {
        ...state,
        isTransferring: true,
      };
    case actionDescriptors.transferInventorySuccessful:
      return {
        ...state,
        isTransferring: false,
      };
    case actionDescriptors.transferInventoryFailed:
      return {
        ...state,
        error: action.error,
        isTransferring: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistory:
      return {
        ...state,
        isInventoryOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchInventoryOwnershipHistorySuccessful:
      return {
        ...state,
        inventoryOwnershipHistory: action.payload,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryDetail:
      return {
        ...state,
        isInventoryDetailsLoading: true,
      };
    case actionDescriptors.fetchInventoryDetailSuccessful:
      return {
        ...state,
        inventoryDetails: action.payload,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.fetchInventoryDetailFailed:
      return {
        ...state,
        error: action.error,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.uploadImage:
      return {
        ...state,
        isUploadImageSubmitting: true,
      };
    case actionDescriptors.uploadImageSuccessful:
      return {
        ...state,
        uploadedImg: action.payload,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.uploadImageFailed:
      return {
        ...state,
        error: action.error,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.createItem:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createItemSuccessful:
      return {
        ...state,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createItemFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchPriceHistory:
      return {
        ...state,
        isFetchingPriceHistory: true,
      };
    case actionDescriptors.fetchPriceHistorySuccessful:
      return {
        ...state,
        isFetchingPriceHistory: false,
        priceHistory: action.payload,
      };
    case actionDescriptors.fetchPriceHistoryFailed:
      return {
        ...state,
        isFetchingPriceHistory: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
