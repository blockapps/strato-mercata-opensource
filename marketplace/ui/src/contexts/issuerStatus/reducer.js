import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.requestReview:
      return {
        ...state,
        requestingReview: true,
      };
    case actionDescriptors.requestReviewSuccessful:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.requestReviewFailed:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.authorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.authorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.authorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.deauthorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.modifyAdmin:
      return {
        ...state,
        changingAdminStatus: true,
      };
    case actionDescriptors.modifyAdminSuccessful:
      return {
        ...state,
        changingAdminStatus: false,
      };
    case actionDescriptors.modifyAdminFailed:
      return {
        ...state,
        changingAdminStatus: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
