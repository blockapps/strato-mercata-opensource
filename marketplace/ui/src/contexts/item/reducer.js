import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchItem:
      return {
        ...state,
        isItemsLoading: true,
      };
    case actionDescriptors.fetchItemSuccessful:
      return {
        ...state,
        items: action.payload,
        isItemsLoading: false,
      };
    case actionDescriptors.fetchItemFailed:
      return {
        ...state,
        error: action.error,
        isItemsLoading: false,
      };
    case actionDescriptors.fetchSerialNumbers:
      return {
        ...state,
        isSerialNumbersLoading: true,
      };
    case actionDescriptors.fetchSerialNumbersSuccessful:
      return {
        ...state,
        serialNumbers: action.payload,
        isSerialNumbersLoading: false,
      };
    case actionDescriptors.fetchSerialNumbersFailed:
      return {
        ...state,
        error: action.error,
        isSerialNumbersLoading: false,
      };
    case actionDescriptors.fetchItemOwnershipHistory:
      return {
        ...state,
        isOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchItemOwnershipHistorySuccessful:
      return {
        ...state,
        ownershipHistory: action.payload,
        isOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchItemOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemRawMaterials:
      return {
        ...state,
        isRawMaterialsLoading: true,
      };
    case actionDescriptors.fetchItemRawMaterialsSuccessful:
      return {
        ...state,
        rawMaterials: action.payload,
        isRawMaterialsLoading: false,
      };
    case actionDescriptors.fetchItemRawMaterialsFailed:
      return {
        ...state,
        error: action.error,
        isRawMaterialsLoading: false,
      };
    case actionDescriptors.setActualRawMaterials:
      return {
        ...state,
        actualRawMaterials: action.payload,
      };
    case actionDescriptors.transferOwnership:
      return {
        ...state,
        isTransferringItems: true,
      };
    case actionDescriptors.transferOwnershipSuccessful:
      return {
        ...state,
        isTransferringItems: false,
      };
    case actionDescriptors.transferOwnershipFailed:
      return {
        ...state,
        error: action.error,
        isTransferringItems: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
