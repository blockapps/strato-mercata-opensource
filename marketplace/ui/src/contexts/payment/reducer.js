import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.getPaymentServices:
      return {
        ...state,
        arePaymentServicesLoading: true,
      };
    case actionDescriptors.getPaymentServicesSuccessful:
      return {
        ...state,
        paymentServices: action.payload.data,
        paymentServicesTotal: action.payload.count,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getPaymentServicesFailed:
      return {
        ...state,
        error: action.error,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getNotOnboarded:
      return {
        ...state,
        areNotOnboardedLoading: true,
      };
    case actionDescriptors.getNotOnboardedSuccessful:
      return {
        ...state,
        notOnboarded: action.payload.data,
        notOnboardedTotal: action.payload.count,
        areNotOnboardedLoading: false,
      };
    case actionDescriptors.getNotOnboardedFailed:
      return {
        ...state,
        error: action.error,
        areNotOnboardedLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
