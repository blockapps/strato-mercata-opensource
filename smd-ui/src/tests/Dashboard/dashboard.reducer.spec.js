import reducer from '../../components/Dashboard/dashboard.reducer';
import {
  preloadBlockNumber,
  updateBlockNumber,
  preloadContractCount,
  updateContractCount,
  preloadUsersCount,
  updateUsersCount,
  preloadTransactionsCount,
  updateTransactionCount,
  preloadBlockDifficulty,
  updateBlockDifficulty,
  preloadBlockPropagation,
  updateBlockPropagation,
  preloadTransactionType,
  updateTransactionType
} from '../../components/Dashboard/dashboard.action';
import { dashboard, initialState } from './dashboardMock';

describe('Dashboard: reducer', () => {

  // INITIAL_STATE
  test('set initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  // BLOCK_NUMBER
  describe('BLOCK_NUMBER:', () => {
    // PRELOAD_BLOCK_NUMBER
    test('store', () => {
      const action = preloadBlockNumber(dashboard.lastBlockNumber);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // UPDATE_BLOCK_NUMBER
    test('update', () => {
      let lastBlockNumber = 2;
      const action = updateBlockNumber(lastBlockNumber);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });
  });

  // CONTRACT_COUNT
  describe('CONTRACT_COUNT:', () => {
    // PRELOAD_CONTRACT_COUNT
    test('store', () => {
      const action = preloadContractCount(dashboard.contractsCount);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // UPDATE_CONTRACT_COUNT
    test('update', () => {
      let contractsCount = 4;
      const action = updateContractCount(contractsCount);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });
  });

  // USERS_COUNT
  describe('USERS_COUNT:', () => {
    // PRELOAD_USERS_COUNT
    test('store', () => {
      const action = preloadUsersCount(dashboard.usersCount);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // UPDATE_USERS_COUNT
    test('update', () => {
      let usersCount = 2;
      const action = updateUsersCount(usersCount);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });
  });

  // TRANSACTION_COUNT
  describe('TRANSACTION_COUNT:', () => {
    // PRELOAD_TRANSACTION_COUNT
    test('store', () => {
      const action = preloadTransactionsCount(dashboard.transactionsCount);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // UPDATE_TRANSACTION_COUNT
    test('update', () => {
      let transactionsCount = [{
        "x": 0,
        "y": 0
      }];
      const action = updateTransactionCount(transactionsCount);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });
  });

  // BLOCK_DIFFICULTY
  describe('BLOCK_DIFFICULTY:', () => {
    // PRELOAD_BLOCK_DIFFICULTY
    test('store', () => {
      const action = preloadBlockDifficulty(dashboard.blockDifficulty);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // UPDATE_BLOCK_DIFFICULTY
    test('update', () => {
      let blockDifficulty = [{
        "x": 0,
        "y": 8
      }];
      const action = updateBlockDifficulty(blockDifficulty);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });
  });

  // BLOCK_PROPAGATION
  describe('BLOCK_PROPAGATION:', () => {
    // PRELOAD_BLOCK_PROPAGATION
    test('store', () => {
      const action = preloadBlockPropagation(dashboard.blockPropagation);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // UPDATE_BLOCK_PROPAGATION
    test('update', () => {
      let blockPropagation = [{
        "x": 0,
        "y": 3358
      }];
      const action = updateBlockPropagation(blockPropagation);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });
  });

  // TRANSACTION_TYPES
  describe('TRANSACTION_TYPES:', () => {
    // PRELOAD_TRANSACTION_TYPES
    test('store', () => {
      const action = preloadTransactionType(dashboard.transactionTypes);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // UPDATE_TRANSACTION_TYPES
    test('update', () => {
      let transactionTypes = [{
        "val": 1,
        "type": "Transfer"
      },
      {
        "val": 2,
        "type": "Contract"
      }];
      const action = updateTransactionType(transactionTypes);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });
  });

});