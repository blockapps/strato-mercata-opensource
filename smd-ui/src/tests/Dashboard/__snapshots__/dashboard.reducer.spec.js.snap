// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard: reducer BLOCK_DIFFICULTY: store 1`] = `
Object {
  "blockDifficulty": Array [
    Object {
      "x": 0,
      "y": 8192,
    },
    Object {
      "x": 1,
      "y": 131072,
    },
    Object {
      "x": 2,
      "y": 131072,
    },
    Object {
      "x": 3,
      "y": 131072,
    },
    Object {
      "x": 4,
      "y": 131072,
    },
    Object {
      "x": 5,
      "y": 131072,
    },
    Object {
      "x": 6,
      "y": 131072,
    },
    Object {
      "x": 7,
      "y": 131072,
    },
    Object {
      "x": 8,
      "y": 131072,
    },
    Object {
      "x": 9,
      "y": 131072,
    },
  ],
  "blockPropagation": Array [],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer BLOCK_DIFFICULTY: update 1`] = `
Object {
  "blockDifficulty": Array [
    Object {
      "x": 0,
      "y": 8,
    },
  ],
  "blockPropagation": Array [],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer BLOCK_NUMBER: store 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 9,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer BLOCK_NUMBER: update 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 2,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer BLOCK_PROPAGATION: store 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [
    Object {
      "x": 0,
      "y": 0,
    },
    Object {
      "x": 1,
      "y": 1512994588,
    },
    Object {
      "x": 2,
      "y": 3358,
    },
    Object {
      "x": 3,
      "y": 19,
    },
    Object {
      "x": 4,
      "y": 1846,
    },
    Object {
      "x": 5,
      "y": 301,
    },
    Object {
      "x": 6,
      "y": 48,
    },
    Object {
      "x": 7,
      "y": 55429,
    },
    Object {
      "x": 8,
      "y": 106195,
    },
    Object {
      "x": 9,
      "y": 58,
    },
  ],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer BLOCK_PROPAGATION: update 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [
    Object {
      "x": 0,
      "y": 3358,
    },
  ],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer CONTRACT_COUNT: store 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [],
  "contractsCount": 2,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer CONTRACT_COUNT: update 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [],
  "contractsCount": 4,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer TRANSACTION_COUNT: store 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [
    Object {
      "x": 0,
      "y": 0,
    },
    Object {
      "x": 1,
      "y": 1,
    },
    Object {
      "x": 2,
      "y": 1,
    },
    Object {
      "x": 3,
      "y": 1,
    },
    Object {
      "x": 4,
      "y": 1,
    },
    Object {
      "x": 5,
      "y": 1,
    },
    Object {
      "x": 6,
      "y": 1,
    },
    Object {
      "x": 7,
      "y": 1,
    },
    Object {
      "x": 8,
      "y": 1,
    },
    Object {
      "x": 9,
      "y": 1,
    },
  ],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer TRANSACTION_COUNT: update 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [
    Object {
      "x": 0,
      "y": 0,
    },
  ],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer TRANSACTION_TYPES: store 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [
    Object {
      "type": "FunctionCall",
      "val": 6,
    },
    Object {
      "type": "Transfer",
      "val": 1,
    },
    Object {
      "type": "Contract",
      "val": 2,
    },
  ],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer TRANSACTION_TYPES: update 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [
    Object {
      "type": "Transfer",
      "val": 1,
    },
    Object {
      "type": "Contract",
      "val": 2,
    },
  ],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 0,
}
`;

exports[`Dashboard: reducer USERS_COUNT: store 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 1,
}
`;

exports[`Dashboard: reducer USERS_COUNT: update 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [],
  "contractsCount": 0,
  "healthStatus": false,
  "lastBlockNumber": 0,
  "shardCount": 0,
  "transactionTypes": Array [],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 2,
}
`;

exports[`Dashboard: reducer set initial state 1`] = `
Object {
  "blockDifficulty": Array [],
  "blockPropagation": Array [],
  "contractsCount": 0,
  "healthStatus": false,
  "ifHovering": false,
  "lastBlockNumber": 0,
  "networkStatus": false,
  "shardCount": 0,
  "systemStatus": false,
  "systemWarnings": "",
  "transactionTypes": Array [],
  "transactionsCount": Array [],
  "uptime": 0,
  "usersCount": 0,
}
`;
